java/code/src/com/redhat/rhn/manager/errata/ErrataManager.java

```diff
@@ -1858,10 +1858,12 @@ private static List<Long> scheduleErrataActions(User scheduler, Org org, Date ea
         // 2.1- compute a system to errata map for minions
         // those get one Action per system, with all erratas in it
         Map<Long, List<Long>> minionErrataMap = minions.stream()
-            .collect(toMap(
-                sid -> sid,
-                sid -> new ArrayList<>(serverErrataMap.get(sid))
-            ));
+                .collect(toMap(
+                        sid -> sid,
+                        sid -> new ArrayList<>(Optional.of(serverErrataMap)
```

Comment:
I don't think `serverErrataMap` can be null here. It's already used on line 1822.

Reply:
the problem is not `serverErrataMap`  but `serverErrataMap.get(sid)`. With the original code I have this failure in unit test

```
java.lang.NullPointerException: Cannot invoke "java.util.Collection.toArray()" because "c" is null
```

Reply:
I see. Then I think `serverErrataMap.getOrDefault(sid, List.of())` would be more readable.